client/src/visitors/pages/qrcode.js

```diff
@@ -136,9 +138,18 @@ export default class QRCode extends Component {
   changeActivity = (newActivity) => {
     this.setState({ activity: newActivity });
```

Comment:
If I'm reading this correctly... The component that fires this function as an onclick is grabbing the activity value from state, which is being set in this function...

Reply:
Lol, yeah, don't ask me, I didn't write it 🤷🏼‍♂️